BootloaderCorePkg/Stage1B/Stage1B.c

```diff
@@ -627,14 +692,26 @@ ContinueFunc (
   BoardInit (PostTempRamExit);
   AddMeasurePoint (0x2070);
 
-  // Extend External Config Data hash
-  if (MEASURED_BOOT_ENABLED() ) {
+  if (MEASURED_BOOT_ENABLED()) {
     if (GetBootMode() != BOOT_ON_S3_RESUME) {
+        // Extend External Config Data hash
         if (Stage1bParam->ConfigDataHashValid == 1) {
-          TpmExtendConfigData (Stage1bParam->ConfigDataHash,
-            PcdGet8(PcdCompSignHashAlg),
-            (UINT8 *) Stage1bParam->CfgDataAddr,
-            ((CDATA_BLOB *) Stage1bParam->CfgDataAddr)->UsedLength);
+          TpmExtendPcrAndLogEvent ( 1,
+                    (TPMI_ALG_HASH) GetTpmHashAlg(PcdGet32(PcdMeasuredBootHashMask)),

```

Comment:
Use a variable for GetTpmHashAlg(PcdGet32(PcdMeasuredBootHashMask)).
if ConfigDataHashValid == 1 and KeyHashManifestHashValid == 1, the same function can be called twice.

Reply:
Variable can be added for GetTpmHashAlg(PcdGet32(PcdMeasuredBootHashMask)).
TpmExtendPcrAndLogEvent hash be added separately for config data and KeyHashManifestHash extend.